src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java

```diff
@@ -30,22 +30,78 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
-
+import java.util.Map;
+import java.util.Map.Entry;
 import javax.faces.component.UIComponent;
 import javax.faces.context.FacesContext;
 import javax.faces.context.ResponseWriter;
 import javax.faces.convert.ConverterException;
-
+import org.json.JSONObject;
 import org.primefaces.component.api.UICalendar;
 import org.primefaces.component.calendar.BaseCalendarRenderer;
 import org.primefaces.expression.SearchExpressionFacade;
 import org.primefaces.expression.SearchExpressionUtils;
+import org.primefaces.model.datepicker.DateMetaData;
+import org.primefaces.model.datepicker.DateMetaDataModel;
+import org.primefaces.model.datepicker.LazyDateMetaDataModel;
 import org.primefaces.util.CalendarUtils;
 import org.primefaces.util.ComponentUtils;
 import org.primefaces.util.WidgetBuilder;
 
 public class DatePickerRenderer extends BaseCalendarRenderer {
 
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        DatePicker datePicker = (DatePicker) component;
+
+        if (context.getExternalContext().getRequestParameterMap().containsKey(datePicker.getClientId(context))) {
+            encodeDateMetaData(context, datePicker);
+        }
+        else {
+            super.encodeEnd(context, component);
+        }
+    }
+
+    protected void encodeDateMetaData(FacesContext context, DatePicker datePicker) throws IOException {
+        String clientId = datePicker.getClientId(context);
+        DateMetaDataModel model = datePicker.getModel();
+        Map<String, String> params = context.getExternalContext().getRequestParameterMap();
+
+        if (model instanceof LazyDateMetaDataModel) {
+            int year = Integer.valueOf(params.get(clientId + "_year"));
+            int month = Integer.valueOf(params.get(clientId + "_month")) + 1;
+
+            LocalDate startDate = LocalDate.of(year, month, 1);
+            LocalDate endDate = startDate.plusMonths(datePicker.getNumberOfMonths()).minusDays(1);
+
+            LazyDateMetaDataModel lazyModel = ((LazyDateMetaDataModel) model);
+            lazyModel.clear();
+            lazyModel.loadDateMetaData(startDate, endDate);
+        }
+
+        encodeDateMetaDataAsJSON(context, datePicker, model);
+    }
+
+    protected void encodeDateMetaDataAsJSON(FacesContext context, DatePicker datePicker, DateMetaDataModel model) throws IOException {
+        JSONObject jsonDateMetaData = new JSONObject();
```

Comment:
Rather than checking mode != null 2 lines down why not make the first line of this method `if (model == null) return;` to short circuit this whole method?